src/UI/SubtitleEntry.lua

```diff
@@ -50,7 +50,7 @@ function SubtitleEntry.new(Message: string, Window: Types.SubtitleWindow, Refere
     TextLabel.ZIndex = 2
     TextLabel.Parent = Window.RowAdornFrame
     self.TextLabel = TextLabel
-    table.insert(Window.SubtitleEntries, self)
+    table.insert(Window.SubtitleEntries, self :: any)
```

Comment:
`self` is a table, I see no reason why it should be typed as `any` instead of using `{}`. Perhaps making this a custom type will be better.

Reply:
For all of these, it is because the types (like `Types.SubtitleEntry`) don't contain the metatable that `self` has. In my other projects, I use `(self :: any) :: SomeType`, which isn't great. Metatables aren't that great for typing in general with Luau.

Reply:
Alright, didn't know that. Thanks for explaining!